### ISSUE TITLE
Channel names in uncompressed binary file are truncated to 10 characters

### ISSUE THREAD

**Issue by ebranlard** [OpenFAST/openfast#1812]
**Main issue**
Uncompressed binaries use only 10 characters for the channel names when `OutFmt = 4` i.e. `FileID = FileFmtID_NoCompressWithoutTime = 3` , see e.g: 

https://github.com/OpenFAST/openfast/blob/df840357b903877309eaa4d35e8b2529de3ab1d8/modules/nwtc-library/src/NWTC_IO.f90#L5011

https://github.com/OpenFAST/openfast/blob/df840357b903877309eaa4d35e8b2529de3ab1d8/modules/nwtc-library/src/NWTC_IO.f90#L6883

**Describe the solution you'd like**
The maximum number of characters for the channel length should be written to the file to have a similar behavior for compressed and uncompressed binaries, and ASCII files. 

Since uncompressed binaries are not too commonly used at the moment, I suggest that when` OutFmt = 4`  OpenFAST writes an uncompressed binary file with one extra field for the channel length (similar to what is done for `FileFmtID_ChanLen_In`). 
For backward compatbility, we can introduce a new "FileID" in the binary file,  e.g. `FileID = FileFmtID_NoCompressWithoutTime_ChanLen_In=5` .

**Comment by ebranlard** [comment]
Compressed binaries can perform poorly for signals that have a large range, typically during initial transients that are slightly unphysical. This can be avoided by starting the output after the transients, or using a steady state solve, or having better initial conditions. Since not so many people use this format, I think it's a small change to add the channel length as an extra field in this file format. 
Below is an example from a CFD simulation (in log scale) where the compressed binary does not work well for the Cp because it takes large values at initialization:
![image](https://github.com/OpenFAST/openfast/assets/1318316/9c901a6f-848f-46fc-9015-fd74c8d3a8aa)
We could also bound these coefficients between -10 and 10 for instance...

**Comment by bjonkman** [comment]
I've seen something similar in the Cp output in some of the extreme wind-direction change cases, but I've also been trying to bound that coefficient in my code. If 10^4 was a valid value, then the differences in those smaller values due to compression wouldn't matter. :) 

It's not a huge deal to add another format in OpenFAST, it's just more bookkeeping on the file reading side. There tend to be a bunch of different post-processors where the change would need to be made.

**Comment by andrew-platt** [comment]
Is this still an issue?

**Comment by bjonkman** [comment]
Yes. The uncompressed binary file format has not been changed, and most of the r-test cases still output uncompressed binary formats (it looks like AeroDyn driver and FAST.Farm use compressed output)

**Comment by andrew-platt** [comment]
This is definitely something to fix at some point... maybe in v4.0.1

**Comment by bjonkman** [comment]
Since it will change the header of the uncompressed binary file, it's going to affect all of the file-read codes. I'd either (1) put it in v5.0.0 or (2) modify r-test to use the compressed files (which have a field for channel name width) and leave this as a known limitation for a file type that is rarely used.

**Comment by andrew-platt** [comment]
Very good point.  We'll put it in 5.0 (and maybe 4.1 if we have such a release).
